### PR TITLE
GUVNOR-3200: [Guided Decision Table] Access control to prohibit business users from modifying columns

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.jboss.org/browse/GUVNOR-3200

---

permissoes

---

Part of an ensemble:
- https://github.com/AppFormer/uberfire/pull/861
- https://github.com/kiegroup/kie-wb-common/pull/1190
- https://github.com/kiegroup/drools-wb/pull/636
- https://github.com/kiegroup/kie-wb-distributions/pull/628
- https://github.com/kiegroup/kie-docs/pull/402

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/WorkbenchFeatures.java [hunk 1]
```diff
@@ -26,4 +26,6 @@
     String PLANNER_AVAILABLE = "planner.available";
 
     String EDIT_GLOBAL_PREFERENCES = "globalpreferences.edit";
-}
\ No newline at end of file
+
+    String EDIT_GUIDED_DECISION_TABLE = "guideddecisiontable.edit";
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/WorkbenchFeatures.java]: Following our discussion on the related PR it's probably better to rename this (something like) ```guideddecisiontable.edit.columns```

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/WorkbenchTreeProvider.java [hunk 2]
```diff
@@ -103,12 +103,16 @@ public void loadChildren(PermissionNode parent,
         PermissionLeafNode node5 = createPermissionLeafNode(EDIT_GLOBAL_PREFERENCES,
                                                             i18n.EditGlobalPreferences(),
                                                             i18n.EditGlobalPreferencesHelp());
+        PermissionLeafNode node6 = createPermissionLeafNode(EDIT_GUIDED_DECISION_TABLE,
+                                                            i18n.EditGuidedDecisionTable(),
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/WorkbenchTreeProvider.java]: And probably best to make these i18n's refer more explicitly to **columns** too.

**Comment by karreiro** [comment]
@manstis @jomarko Changes applied. Could you please take a look? :-)

**Comment by manstis** [comment]
@karreiro I would encourage you to also check the ```kie-wb-distributions``` module does not need changes to the permissions setup (IDK whether it does or does not BTW!)

**Comment by karreiro** [comment]
@manstis nice catch! Thanks!

**Pull request merged**